rclcpp/src/rclcpp/time_source.cpp

```diff
@@ -185,14 +185,21 @@ void TimeSource::set_clock(
   const builtin_interfaces::msg::Time::SharedPtr msg, bool set_ros_time_enabled,
   std::shared_ptr<rclcpp::Clock> clock)
 {
+  std::lock_guard<std::mutex> clock_guard(clock->get_clock_mutex());
+
+  rcl_ret_t ret = RCL_RET_OK;
   // Do change
   if (!set_ros_time_enabled && clock->ros_time_is_active()) {
-    disable_ros_time(clock);
+    ret = rcl_disable_ros_time_override(clock->get_clock_handle());
   } else if (set_ros_time_enabled && !clock->ros_time_is_active()) {
-    enable_ros_time(clock);
+    ret = rcl_enable_ros_time_override(clock->get_clock_handle());
+  }
+  if (ret != RCL_RET_OK) {
+    rclcpp::exceptions::throw_from_rcl_error(
+      ret, "Failed to enable ros_time_override_status");
```

Comment:
Nit: In the case we disabled it, it should be:

`Failed to disable ros_time_override_status`

(this was wrong in the previous code too)

Reply:
fixed!

Reply:
Looks good, thanks.